### PR TITLE
loader 0.15.0

### PR DESCRIPTION
**Pull request by tildejustin**

### CODE REVIEW

File: generate.py [hunk 1]
```diff
@@ -11,59 +12,38 @@ class IntermediaryType(StrEnum):
     Ornithe = "net.fabricmc.intermediary.ornithe.pre-1.6.json"
 
 
-def mkdir_if_not_exists(path: str):
-    if not os.path.exists(path):
-        os.mkdir(path)
+def mkdir(*paths: str):
```
Comment (DuncanRuns) [inline generate.py]: prolly call this `mkdirs` since it can make multiple dir

File: generate.py [hunk 2]
```diff
@@ -72,9 +52,24 @@ def cleanup(self):
                 os.remove(os.path.join(root, file))
             for directory in dirs:
                 os.rmdir(os.path.join(root, directory))
-
         os.rmdir(self.path)
 
+    def process_file(self, *files: str, out: Optional[str] = None):
```
Comment (DuncanRuns) [inline generate.py:57]: the default parameter (`= None`) already implies optional, `Optional` is nice and all but does end up adding another import ¯\\\_(ツ)\_/¯

Also `files` is a list[str] right

File: generate.py [hunk 3]
```diff
@@ -93,11 +88,11 @@ def cleanup(self):
     ("1.0", "2.9.0", IntermediaryType.Ornithe)
 ]
 
-loader = "0.14.24"
-mkdir_if_not_exists("out")
+loader_version = "0.15.0"
```
Comment (DuncanRuns) [inline generate.py]: could use naming style for constants (`LOADER_VERSION`)

**Review by DuncanRuns — APPROVED** [review]
comments of this review are some silly nits

the pr is more than "loader 0.15.0" and involves some refactors, but the file is small and the generation involves almost no input so if it generates files correctly and it works, then it works 👍

File: generate.py (hunk 2 above)
Comment (tildejustin) [reply generate.py:57]: you're supposed to type *\<vars\> with their element type

**Pull request merged**